src/components/Workspace.jsx

```diff
@@ -128,6 +131,24 @@ class Workspace extends React.Component {
     addEventListener('beforeunload', this._confirmUnload);
   }
 
+  componentDidUpdate(prevProps) {
```

Comment:
Yikes, good catch on the issues with `react-popout`. Well, I think it should be pretty straightforward to hand-roll something that does what we need!